backend/data/src/main/kotlin/io/tolgee/service/NotificationService.kt

```diff
@@ -0,0 +1,21 @@
+package io.tolgee.service
+
+import io.tolgee.repository.NotificationRepository
+import org.springframework.stereotype.Service
+
+@Service
+class NotificationService(
+  private val notificationRepository: NotificationRepository,
+) {
+  fun getNotifications(userId: Long): List<NotificationModel> {
+    return notificationRepository.fetchNotificationsByUserId(userId)
+      .map { NotificationModel(it.id, it.linkedTask?.project?.id, it.linkedTask?.number, it.linkedTask?.name) }
+  }
+}
+
+data class NotificationModel(
+  val id: Long,
+  val linkedProjectId: Long?,
+  val linkedTaskNumber: Long?,
+  val linkedTaskName: String?,
```

Comment:
Provide the task using the TaskModelAssembler and TaskModel. Otherwise we will provide many different shapes in the API, which we don't want. Also that way you can re-use the existing model assemblers.

Reply:
Also the Models ahould be always created using their model assemblers in separate according to the conventions.